src/Pause.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Resque\Plugins;
+
+use Resque;
+use Resque_Event;
+use Resque_Job_DontCreate;
+
+/**
+ * Base Resque Pause class
+ *
+ * @package        PHP Resque Pause
+ * @author         Wedy Chainy <askwedi@wedipedia.org>
+ * @licence        http://www.opensource.org/licenses/mit-license.php
+ */
+class Pause
+{
+    /** @var JobPauser  */
+    private $pauser = null;
+
+    /** @var JobPauser  */
+    private $listener = null;
+
+    public function __construct()
+    {
+        // Create object for pausing jobs
+        $this->pauser = new JobPauser(Resque::redis(), \Resque_Redis::getPrefix());
+        $pause = $this;
+        $pauser = $this->pauser;
+
+        // Listen for enqueue and move any new jobs to temp queue
+        $this->listener = function ($class, $args, $queue, $id) use (&$pause, $pauser) {
+            if ($pause->isPaused($queue)) {
```

Comment:
This is the one place where `$this` is used here. I'm considering switching it for the underlying `$pauser` function which had to be passed in for the `pushPausedJob` method anyway. Opinions?